cdap-common/src/main/java/io/cdap/cdap/common/utils/TimeBucketHasher.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.utils;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class for generating time bucketed hashes
+ */
+public class TimeBucketHasher {
```

Comment:
Since there is only one static method in this class and only used in one place, I think it's better to be a private static method in the `InMemoryProgramRunDispatcher` class. You can have it VisibleForTesting to make the testing easier.

Reply:
Done